src/app/item-popup/ItemStat.tsx

```diff
@@ -213,9 +214,10 @@ function breakDownTotalValue(statValue: number, item: D2Item) {
       if (socket.plug && socket.plug.stats) {
         for (const statHash of armorStats) {
           const plugStatValue = socket.plug.stats[statHash];
+
           if (plugStatValue) {
             baseTotalValue -= plugStatValue;
-            if (socket.plug.isMasterwork) {
+            if (socket.plug.stats[MASTERWORK_STAT_HASH] === 10) {
```

Comment:
Could you maybe check to see if the item category is "masterwork mods"? https://data.destinysets.com/i/ItemCategory:141186804

I feel like this should be able to handle both Armor 2.0 and Armor 1.0 the same way.

Reply:
The item category isn't masterwork mods, its armor upgrade, https://data.destinysets.com/i/SocketCategory:760375309

Here is the example from the json @sundevour commented earlier.
<details><summary>Category for masterwork</summary>

```
"categories": [
      {
        "category": {
          "displayProperties": {
            "description": "The energy type and tier of a piece of armor determines which armor mods you can attach to it. Upgrade its armor tier to increase its capacity for more expensive armor mods.",
            "name": "ARMOR TIER",
            "hasIcon": false
          },
          "uiCategoryStyle": 750616615,
          "categoryStyle": 5,
          "hash": 760375309,
          "index": 2,
          "redacted": false,
          "blacklisted": false
        },
        "sockets": [
          {
            "socketIndex": 5,
            "plug": {
              "plugItem": {
                "displayProperties": {
                  "description": "Upgrade this armor to increase its energy capacity by 1. Fully upgrading this armor to Masterwork provides a bonus to all stats.\n\nThis armor cannot accept mods that require Arc or Solar energy.",
                  "name": "Upgrade Armor",
                  "icon": "/common/destiny2_content/icons/648be777f7416bf8080c4a30af224626.jpg",
                  "hasIcon": true
                },
                "tooltipNotifications": [],
                "backgroundColor": {
                  "colorHash": 0,
                  "red": 0,
                  "green": 0,
                  "blue": 0,
                  "alpha": 0
                },
                "itemTypeDisplayName": "",
                "uiItemDisplayStyle": "",
                "itemTypeAndTierDisplayName": "",
                "displaySource": "",
                "tooltipStyle": "vendor_action",
                "action": {
                  "verbName": "Dismantle",
                  "verbDescription": "",
                  "isPositive": true,
                  "requiredCooldownSeconds": 0,
                  "requiredItems": [],
                  "progressionRewards": [],
                  "rewardSheetHash": 0,
                  "rewardItemHash": 0,
                  "rewardSiteHash": 0,
                  "requiredCooldownHash": 0,
                  "deleteOnAction": true,
                  "consumeEntireStack": false,
                  "useOnAcquire": false
                },
                "inventory": {
                  "maxStackSize": 1,
                  "bucketTypeHash": 3313201758,
                  "recoveryBucketTypeHash": 215593132,
                  "tierTypeHash": 4008398120,
                  "isInstanceItem": false,
                  "nonTransferrableOriginal": false,
                  "tierTypeName": "Legendary",
                  "tierType": 5,
                  "expirationTooltip": "",
                  "expiredInActivityMessage": "",
                  "expiredInOrbitMessage": "",
                  "suppressExpirationWhenObjectivesComplete": true
                },
                "sourceData": {
                  "exclusive": 0,
                  "vendorSources": []
                },
                "plug": {
                  "insertionRules": [],
                  "plugCategoryIdentifier": "v460.plugs.armor.masterworks.stat.resistance_4",
                  "plugCategoryHash": 400813045,
                  "onActionRecreateSelf": false,
                  "actionRewardSiteHash": 0,
                  "actionRewardItemOverrideHash": 0,
                  "insertionMaterialRequirementHash": 3588361558,
                  "previewItemOverrideHash": 0,
                  "enabledMaterialRequirementHash": 0,
                  "enabledRules": [],
                  "uiPlugLabel": "masterwork",
                  "plugStyle": 1,
                  "plugAvailability": 2,
                  "alternateUiPlugLabel": "",
                  "alternatePlugStyle": 0,
                  "isDummyPlug": false,
                  "energyCapacity": {
                    "capacityValue": 10,
                    "energyTypeHash": 4069572561,
                    "energyType": 3
                  }
                },
                "acquireRewardSiteHash": 0,
                "acquireUnlockHash": 0,
                "investmentStats": [
                  {
                    "statTypeHash": 16120457,
                    "value": 10,
                    "isConditionallyActive": false
                  },
                  {
                    "statTypeHash": 2996146975,
                    "value": 2,
                    "isConditionallyActive": false
                  },
                  {
                    "statTypeHash": 392767087,
                    "value": 2,
                    "isConditionallyActive": false
                  },
                  {
                    "statTypeHash": 1735777505,
                    "value": 2,
                    "isConditionallyActive": false
                  },
                  {
                    "statTypeHash": 144602215,
                    "value": 2,
                    "isConditionallyActive": false
                  },
                  {
                    "statTypeHash": 1943323491,
                    "value": 2,
                    "isConditionallyActive": false
                  },
                  {
                    "statTypeHash": 4244567218,
                    "value": 2,
                    "isConditionallyActive": false
                  }
                ],
                "perks": [],
                "allowActions": true,
                "doesPostmasterPullHaveSideEffects": false,
                "nonTransferrable": false,
                "itemCategoryHashes": [
                  59
                ],
                "specialItemType": 0,
                "itemType": 19,
                "itemSubType": 0,
                "classType": 3,
                "breakerType": 0,
                "equippable": false,
                "defaultDamageType": 0,
                "isWrapper": false,
                "hash": 2765543780,
                "index": 8676,
                "redacted": false,
                "blacklisted": false
              },
              "enabled": true,
              "enableFailReasons": "",
              "plugObjectives": [],
              "perks": [],
              "isMasterwork": false,
              "stats": {
                "16120457": 10,
                "144602215": 2,
                "392767087": 2,
                "1735777505": 2,
                "1943323491": 2,
                "2996146975": 2,
                "4244567218": 2
              }
            },
            "plugOptions": [
              {
                "plugItem": {
                  "displayProperties": {
                    "description": "Upgrade this armor to increase its energy capacity by 1. Fully upgrading this armor to Masterwork provides a bonus to all stats.\n\nThis armor cannot accept mods that require Arc or Solar energy.",
                    "name": "Upgrade Armor",
                    "icon": "/common/destiny2_content/icons/648be777f7416bf8080c4a30af224626.jpg",
                    "hasIcon": true
                  },
                  "tooltipNotifications": [],
                  "backgroundColor": {
                    "colorHash": 0,
                    "red": 0,
                    "green": 0,
                    "blue": 0,
                    "alpha": 0
                  },
                  "itemTypeDisplayName": "",
                  "uiItemDisplayStyle": "",
                  "itemTypeAndTierDisplayName": "",
                  "displaySource": "",
                  "tooltipStyle": "vendor_action",
                  "action": {
                    "verbName": "Dismantle",
                    "verbDescription": "",
                    "isPositive": true,
                    "requiredCooldownSeconds": 0,
                    "requiredItems": [],
                    "progressionRewards": [],
                    "rewardSheetHash": 0,
                    "rewardItemHash": 0,
                    "rewardSiteHash": 0,
                    "requiredCooldownHash": 0,
                    "deleteOnAction": true,
                    "consumeEntireStack": false,
                    "useOnAcquire": false
                  },
                  "inventory": {
                    "maxStackSize": 1,
                    "bucketTypeHash": 3313201758,
                    "recoveryBucketTypeHash": 215593132,
                    "tierTypeHash": 4008398120,
                    "isInstanceItem": false,
                    "nonTransferrableOriginal": false,
                    "tierTypeName": "Legendary",
                    "tierType": 5,
                    "expirationTooltip": "",
                    "expiredInActivityMessage": "",
                    "expiredInOrbitMessage": "",
                    "suppressExpirationWhenObjectivesComplete": true
                  },
                  "sourceData": {
                    "exclusive": 0,
                    "vendorSources": []
                  },
                  "plug": {
                    "insertionRules": [],
                    "plugCategoryIdentifier": "v460.plugs.armor.masterworks.stat.resistance_4",
                    "plugCategoryHash": 400813045,
                    "onActionRecreateSelf": false,
                    "actionRewardSiteHash": 0,
                    "actionRewardItemOverrideHash": 0,
                    "insertionMaterialRequirementHash": 3588361558,
                    "previewItemOverrideHash": 0,
                    "enabledMaterialRequirementHash": 0,
                    "enabledRules": [],
                    "uiPlugLabel": "masterwork",
                    "plugStyle": 1,
                    "plugAvailability": 2,
                    "alternateUiPlugLabel": "",
                    "alternatePlugStyle": 0,
                    "isDummyPlug": false,
                    "energyCapacity": {
                      "capacityValue": 10,
                      "energyTypeHash": 4069572561,
                      "energyType": 3
                    }
                  },
                  "acquireRewardSiteHash": 0,
                  "acquireUnlockHash": 0,
                  "investmentStats": [
                    {
                      "statTypeHash": 16120457,
                      "value": 10,
                      "isConditionallyActive": false
                    },
                    {
                      "statTypeHash": 2996146975,
                      "value": 2,
                      "isConditionallyActive": false
                    },
                    {
                      "statTypeHash": 392767087,
                      "value": 2,
                      "isConditionallyActive": false
                    },
                    {
                      "statTypeHash": 1735777505,
                      "value": 2,
                      "isConditionallyActive": false
                    },
                    {
                      "statTypeHash": 144602215,
                      "value": 2,
                      "isConditionallyActive": false
                    },
                    {
                      "statTypeHash": 1943323491,
                      "value": 2,
                      "isConditionallyActive": false
                    },
                    {
                      "statTypeHash": 4244567218,
                      "value": 2,
                      "isConditionallyActive": false
                    }
                  ],
                  "perks": [],
                  "allowActions": true,
                  "doesPostmasterPullHaveSideEffects": false,
                  "nonTransferrable": false,
                  "itemCategoryHashes": [
                    59
                  ],
                  "specialItemType": 0,
                  "itemType": 19,
                  "itemSubType": 0,
                  "classType": 3,
                  "breakerType": 0,
                  "equippable": false,
                  "defaultDamageType": 0,
                  "isWrapper": false,
                  "hash": 2765543780,
                  "index": 8676,
                  "redacted": false,
                  "blacklisted": false
                },
                "enabled": true,
                "enableFailReasons": "",
                "plugObjectives": [],
                "perks": [],
                "isMasterwork": false,
                "stats": {
                  "16120457": 10,
                  "144602215": 2,
                  "392767087": 2,
                  "1735777505": 2,
                  "1943323491": 2,
                  "2996146975": 2,
                  "4244567218": 2
                }
              }
            ],
            "hasRandomizedPlugItems": false
          }
        ]
      },
```

</details>

TBH I didn't even consider armour 1.0 for this. I will update it to do so though because they are being displayed with stats values. I get it all happening in the one operation.

I am guessing this is going to be an issue with displaying masterworking on the stat bars too and it's going to be interesting what happens to class items when they are masterworked. From my current understanding of the API and bungie saying masterworked class items give a +2 to each stat, I think there is going to be some tiny stat bars once that has happened. 

Do you want me to raise any issues around this? Or until we have some data to properly test it?